src/content/docs/en/reference/routing-reference.mdx

```diff
@@ -0,0 +1,293 @@
+
+There is no separate routing configuration in Astro. Every [supported page file](/en/basics/astro-pages/#supported-page-files) located within the special `src/pages/` directory creates a single page route, or multiple page routes dynamically when the file name contains a [parameter](#params).
+
+
+## `getStaticPaths()`
+
+**Type:** `(options: GetStaticPathsOptions) => Promise<GetStaticPathsResult> | GetStaticPathsResult`
+
+If a page uses dynamic params in the filename, that component will need to export a `getStaticPaths()` function.
+
+This function is required because Astro is a static site builder. That means that your entire site is built ahead of time. If Astro doesn't know to generate a page at build time, your users won't see it when they visit your site.
+
+```astro
+---
+export async function getStaticPaths() {
+  return [
+    { params: { /* required */ }, props: { /* optional */ } },
+    { params: { ... } },
+    { params: { ... } },
+    // ...
+  ];
+}
+---
+<!-- Your HTML template here. -->
+```
+
+The `getStaticPaths()` function should return an array of objects to determine which paths will be pre-rendered by Astro.
+
+It can also be used in static file endpoints for [dynamic routing](/en/guides/endpoints/#params-and-dynamic-routing).
+
+:::tip
+When using TypeScript, use the [`GetStaticPaths`](/en/guides/typescript/#infer-getstaticpaths-types) type utility to ensure type-safe access of your `params` and `props`.
+:::
+
+:::caution
+The `getStaticPaths()` function executes in its own isolated scope once, before any page loads. Therefore you can't reference anything from its parent scope, other than file imports. The compiler will warn if you break this requirement.
+:::
+
+### `params`
+
+The `params` key of every returned object tells Astro what routes to build. The returned params must map back to the dynamic parameters and rest parameters defined in your component filepath.
+
+`params` are encoded into the URL, so only strings are supported as values. The value for each `params` object must match the parameters used in the page name.
+
+For example, suppose that you have a page at `src/pages/posts/[id].astro`. If you export `getStaticPaths` from this page and return the following for paths:
+
+```astro
+---
+export async function getStaticPaths() {
+  return [
+    { params: { id: '1' } },
+    { params: { id: '2' } },
+    { params: { id: '3' } }
+  ];
+}
+
+const { id } = Astro.params;
+---
+<h1>{id}</h1>
+```
+
+Then Astro will statically generate `posts/1`, `posts/2`, and `posts/3` at build time.
+
+### Data Passing with `props`
+
+To pass additional data to each generated page, you can also set a `props` value on every returned path object. Unlike `params`, `props` are not encoded into the URL and so aren't limited to only strings.
+
+For example, suppose that you generate pages based off of data fetched from a remote API. You can pass the full data object to the page component inside of `getStaticPaths`:
```

Comment:
I think "based on" might be simpler?
```suggestion
For example, suppose that you generate pages based on data fetched from a remote API. You can pass the full data object to the page component inside of `getStaticPaths`:
```

Reply:
Agreed it's from. And it's not "suppose that" :smile: